rre-dataset-generator/dataset_generator.py

```diff
@@ -1,27 +1,86 @@
-import logging
-
-from src.config import Config
+# configuration params
 from src.utils import parse_args
+from src.config import Config
+from src.llm.llm_service import LLMService
+from src.llm.llm_config import LLMConfig
 
-from src.search_engine.solr_search_engine import SolrSearchEngine
+# data structures
+from src.search_engine.data_store import DataStore
+
+# build factories
+from src.search_engine import build_search_engine
+from src.llm import build_chat_model
+from src.writers import build_writer
+
+# logging
 from src.logger import configure_logging
 import logging
 
+
 if __name__ == "__main__":
     args = parse_args()
 
     config = Config.load(args.config_file)
+
     if args.verbose:
         configure_logging(logging.DEBUG)
     else:
         configure_logging(logging.INFO)
+    log = logging.getLogger(__name__)
+
+    search_engine = build_search_engine(search_engine_type=config.search_engine_type,
+                                        endpoint=config.search_engine_collection_endpoint)
+    data_store = DataStore()
+
+    user_queries = []
+    if config.queries is not None:
+        with open(config.queries, 'r', encoding='utf-8') as file:
+            for line in file:
+                if line.strip():
+                    user_queries.append(line.strip())
+
+    # retrieval of the documents needed to generate the queries
+    docs_to_generate_queries = search_engine.fetch_for_query_generation(documents_filter=config.documents_filter,
+                                                                        doc_number=config.doc_number,
+                                                                        doc_fields=config.doc_fields)
+    log.debug(f"Number of documents retrieved for generation: {len(docs_to_generate_queries)}")
+    llm = build_chat_model(LLMConfig.load(config.llm_configuration_file))
+    service = LLMService(chat_model=llm)
+
+    num_queries_per_doc = int(( (config.num_queries_needed - len(user_queries)) // config.doc_number) * 1.5)
+
+    # query generation step
+    flag = False
```

Comment:
The name `flag` is not meaningful; I would have preferred something like `all_queries_generated`.

Alternatively, a probably cleaner solution is to write a specific function that returns when all queries have been generated